### PR TITLE
chore(query-plan): Add round-trip tests and fix nil AST handling

### PR DESCRIPTION
**Pull request by chaudum**
**What this PR does / why we need it**:

Add comprehensive round-trip tests for QueryPlan JSON serialization covering log selectors, pipelines, aggregations, binary operations, and complex queries. Also add tests for empty input and nil AST edge cases.

Fix MarshalJSON, MarshalTo, and Size to handle nil AST gracefully instead of panicking. Nil AST now serializes to empty string instead of JSON `null` consistently.

Part of the QueryRequest.Selector to Plan migration.

### CODE REVIEW

File: pkg/querier/plan/plan.go [hunk 1]
```diff
@@ -42,6 +48,9 @@ func (t *QueryPlan) Size() int {
 }
 
 func (t QueryPlan) MarshalJSON() ([]byte, error) {
+	if t.AST == nil {
+		return []byte("null"), nil
+	}
```
[the remark this answers is not in the document]
Comment (chaudum) [reply pkg/querier/plan/plan.go:53]: Updated PR to consistently marshal to empty string for nil Plan

**Review by salvacorts — APPROVED** [review]

**Review by ivkalita — APPROVED** [review]
